core/src/main/java/org/apache/iceberg/util/SortOrderUtil.java

```diff
@@ -74,6 +101,28 @@ public static SortOrder buildSortOrder(Schema schema, PartitionSpec spec, SortOr
     return builder.build();
   }
 
+  private static Map<Pair<Transform<?, ?>, Integer>, PartitionField> requiredClusteringFields(PartitionSpec spec) {
+    Map<Pair<Transform<?, ?>, Integer>, PartitionField> requiredClusteringFields = Maps.newLinkedHashMap();
+    for (PartitionField partField : spec.fields()) {
+      if (!partField.transform().toString().equals("void")) {
+        requiredClusteringFields.put(Pair.of(partField.transform(), partField.sourceId()), partField);
+      }
+    }
+
+    // remove any partition fields that are satisfied by another partition field, like days(ts) and hours(ts)
+    for (PartitionField partField : spec.fields()) {
+      for (PartitionField field : spec.fields()) {
```

Comment:
nit: `field/otherField` or `partField/otherPartField` instead of `partField/field`?

Reply:
Yeah, I can update. It was more for consistency with the other loop, but I think that got moved to a different method because of the complexity.